megamek/src/megamek/server/ServerHelper.java

```diff
@@ -445,12 +445,24 @@ public static void checkAndApplyMagmaCrust(Hex hex, int elevation, Entity entity
                 hex.addTerrain(new Terrain(Terrains.MAGMA, 2));
                 gameManager.sendChangedHex(curPos);
                 for (Entity en : entity.getGame().getEntitiesVector(curPos)) {
-                    gameManager.doMagmaDamage(en, false);
+                    if (en != entity) {
+                        gameManager.doMagmaDamage(en, false);
+                    }
                 }
             }
         }
     }
 
+    /**
+     * Check for movement into magma hex and apply damage. TODO: If first and last step are in magma double it.
+     */
+    public static void checkEnteringMagma(Hex hex, int elevation, Entity entity, GameManager gameManager) {
+
+        if ((hex.terrainLevel(Terrains.MAGMA) == 2) && (elevation == 0) && (entity.getMovementMode() != EntityMovementMode.HOVER)) {
```

Comment:
Does this cause airborne aerospace units to take magma damage?

Reply:
No, the code is a near copy of the crust check code at line 431. I also tested it.